app/components/course-stage-dropdown.ts

```diff
@@ -0,0 +1,37 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import type CourseModel from 'codecrafters-frontend/models/course';
+import type CourseStageModel from 'codecrafters-frontend/models/course-stage';
+import StageModel from 'codecrafters-frontend/models/course-stage';
+
+interface Signature {
+  Element: HTMLDivElement;
+
+  Args: {
+    course: CourseModel;
+    selectedCourseStage: CourseStageModel | null;
+    onAllCourseStagesDropdownLinkClick?: () => void;
+    onRequestedStageChange: (stage: StageModel) => void;
```

Comment:
Why are there two terms for the same thing here? 🤔 How is "selectedCourseStage" and "requestedStage" different? 

We're also using different abbreviations - one says "Stage", other says "CourseStage". Let's stick to a single terminology to be consistent.